ui/src/pages/namespace/Explorer/Tree/components/modals/CreateNew/Page/index.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  DialogClose,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "~/design/Dialog";
+import { PanelTop, PlusCircle } from "lucide-react";
+import { SubmitHandler, useForm } from "react-hook-form";
+
+import Button from "~/design/Button";
+import { FileNameSchema } from "~/api/files/schema";
+import FormErrors from "~/components/FormErrors";
+import Input from "~/design/Input";
+import { addYamlFileExtension } from "../../../../utils";
+import { defaultEndpointFileYaml } from "~/pages/namespace/Explorer/Endpoint/EndpointEditor/utils";
+import { encode } from "js-base64";
+import { useCreateFile } from "~/api/files/mutate/createFile";
+import { useNamespace } from "~/util/store/namespace";
+import { useNavigate } from "react-router-dom";
+import { usePages } from "~/util/router/pages";
+import { useTranslation } from "react-i18next";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+type FormInput = {
+  name: string;
+  fileContent: string;
+};
+
+const NewPage = ({
+  path,
+  close,
+  unallowedNames,
+}: {
+  path?: string;
+  close: () => void;
+  unallowedNames?: string[];
+}) => {
+  const pages = usePages();
+  const { t } = useTranslation();
+  const namespace = useNamespace();
+  const navigate = useNavigate();
+
+  const resolver = zodResolver(
+    z.object({
+      name: FileNameSchema.transform((enteredName) =>
+        addYamlFileExtension(enteredName)
+      ).refine(
+        (nameWithExtension) =>
+          !(unallowedNames ?? []).some(
+            (unallowedName) => unallowedName === nameWithExtension
+          ),
+        {
+          message: t("pages.explorer.tree.newPage.nameAlreadyExists"),
+        }
+      ),
+      fileContent: z.string(),
+    })
+  );
+
+  const {
+    register,
+    handleSubmit,
+    formState: { isDirty, errors, isValid, isSubmitted },
+  } = useForm<FormInput>({
+    resolver,
+    defaultValues: {
+      fileContent: defaultEndpointFileYaml,
+    },
+  });
+
+  const { mutate: createFile, isPending } = useCreateFile({
+    onSuccess: (data) => {
+      /** To Do:
+       * check if we need to update the Notification Bell for the type = page too?
+       *
+       * creating a new workflow might introduce an uninitialized secret.
+       * We need to update the notification bell, to see potential new messages.
+       */
+      // updateNotificationBell();
```

Comment:
Should we delete this comment? I think this TODO: will be forgotten if we don't make a ticket, and I think it is not worth making one for now.